chain/beacon/drand/drand.go

```diff
@@ -201,11 +203,32 @@ func (db *DrandBeacon) VerifyEntry(curr types.BeaconEntry, prev types.BeaconEntr
 	return err
 }
 
-func (db *DrandBeacon) MaxBeaconRoundForEpoch(filEpoch abi.ChainEpoch) uint64 {
+func (db *DrandBeacon) MaxBeaconRoundForEpoch(nv network.Version, filEpoch abi.ChainEpoch) uint64 {
 	// TODO: sometimes the genesis time for filecoin is zero and this goes negative
 	latestTs := ((uint64(filEpoch) * db.filRoundTime) + db.filGenTime) - db.filRoundTime
+
+	if nv <= network.Version15 {
+		return db.maxBeaconRoundV1(latestTs)
+	}
+
+	return db.maxBeaconRoundV2(latestTs)
+}
+
+func (db *DrandBeacon) maxBeaconRoundV1(latestTs uint64) uint64 {
 	dround := (latestTs - db.drandGenTime) / uint64(db.interval.Seconds())
 	return dround
 }
 
+func (db *DrandBeacon) maxBeaconRoundV2(latestTs uint64) uint64 {
+	if latestTs < db.drandGenTime {
+		return 1
```

Comment:
not 0?

Reply:
When `latestTs == db.drandGenTime`, the round will be 1. I don't know if it matters, but it may make more sense to let the round be 0 before that.

Reply:
Not super important for mainnet, but I think your argument makes sense. Changing.

Reply:
Actually, looking at Drand code, round 0 seems to indicate a special value that says "give me the latest entry".

Reply:
https://github.com/drand/drand/blob/master/client/http/http.go#L311-L315

Reply:
Gonna leave as is, for now.